### ISSUE TITLE
Any way to make module to be garbage collected, maybe Module.release()?

### ISSUE THREAD

**Issue by davojan** [laverdet/isolated-vm#247]
It seems like modules inside an isolate are not garbage collected when there are no references left.

Here is a simple reproducible test case (I ran this on node `14.16.0` and isolated-vm `4.2.0` on MacOS Catalina 10.15.7):

```js
import { Isolate } from 'isolated-vm'

async function main(): Promise<void> {
  const isolate = new Isolate({ memoryLimit: 128 })
  const context = await isolate.createContext()

  let module
  for (let i = 0; i < 5; i++) {
    console.log(i)
    module = await isolate.compileModule(`export const arr = new Array(8 * 1000 * 1000)`, { filename: 'test.ts' })
    await module.instantiate(context, spec => { throw new Error(`not supported ${spec}`) })
    await module.evaluate({ timeout: 1000 })
  }
}

main().catch(err => console.error(err))
```

Here intuitively after each for-loop iteration the only reference to module is overwritten and it should be eligible for GC to free up memory for the newly created modules. But the isolate is disposed due to OOM after the third or forth iteration. I even tried to add 10-20 second sleeps between the iterations (supposing that the GC didn't have time to make it's job), but that didn't help.

For me this looks like a bug. But I wonder if I do something wrong? Are instantiated modules supposed to be garbage collected?

**Comment by laverdet** [comment]
The garbage collector for the inner isolate is running (and panicking), but the outer isolate is keeping these handles around which prevents the garbage in the child isolate from being collected. You reset `module` to something else which does indeed allow the resource to be collected, but until the outer nodejs garbage collection runs they will be kept alive behind the scenes. If you run node with `--expose-gc` and add a call to `gc()` to the top of your loop, then the loop can run forever. 

Most other handles have a `.release()` method for this kind of situation but modules don't have that feature at this time. This just isn't a very common use case, and modules aren't super complete to begin with.

**Comment by davojan** [comment]
Thanks for quick reply! All you said totally make sense, I've tried the trick with `global.gc()` and it worked fine. So I can consider this as a workaround, but if you don't mind I would like to leave this issue as a feature request for the more straightforward `Module.release()` API.

Let me try to explain our use case where modules are the primary feature of the *isolated-vm*. We managed to create something like online IDE for untrusted code with lightweight runtime using *isolated-vm* with full power of es6 modules. And it works quite well. Every time developer changes the source code of some module it's new version is loaded and used in the existing isolate and I try to make the obsolete versions to be unloaded from the memory. So it would be very helpful to have `Module.release()` to have more controlled way to achieve that.

Thank you very much for your great work on this library!
